model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend

```diff
@@ -0,0 +1,243 @@
+        "REQ-EXPR-002",
+        "REQ-EXPR-004",
+        "REQ-EXPR-006",
+        "REQ-EXPR-007",
+        "REQ-EXPR-008",
+        "REQ-EXPR-022"
+    ])
+    def void tcs018() {
+        '''
+            model modelTCS018;
+            
+            // Decimal
+            type numeric Decimal(precision = 13, scale = 4);
+            
+            type boolean Boolean;
+            type date Date;
+            type time Time;
+            type timestamp Timestamp;
+            type numeric Integer(precision = 9, scale = 0);
+            type numeric Long(precision = 15, scale = 0);
+            type string String(min-size = 0, max-size = 4000);
+            
+            // my enum
+            enum MyEnum {
+                A01 = 1;
+                A02 = 2;
+                A03 = 3;
+                A00 = 0;
+            }
+            
+            // MyEntity
+            entity MyEntity {
+                field Timestamp fCreated = Timestamp!now();
+                field Boolean   fBool;
+                field Date      fDate;
+                field Time      fTime;
+                field Long      fLong;
+                field String    fString;
+                field Decimal   fDecimal;
+                field MyEnum    fEnum;
+            }
+            
+            // Entity for the results of queries
+            entity Snapshot {
+                field Timestamp created = Timestamp!now();
+            
+                // relations
+                relation MyEntity[] defEntities; // = setOfMyEntities();
+            }
+            
+            query MyEntity[] queryByTimestamp(Timestamp p1 = `2019-07-18T11:11:12.003+02:00`) =>
+                MyEntity!all()
+                !filter(e | e.fCreated <= p1);
+            
+            query MyEntity[] queryByBoolean(Boolean p1 = false) =>
+                MyEntity!all()
+                !filter(e | e.fBool == p1);
+            
+            query MyEntity[] queryByDate(Date p1 = `2019-07-18`) =>
+                MyEntity!all()
+                !filter(e | e.fDate == p1);
+            
+            query MyEntity[] queryByTime(Time p1 = `11:11:12`) =>
+                MyEntity!all()
+                !filter(e | e.fTime == p1);
+            
+            query MyEntity[] queryByLong(Long p1 = 9999) =>
+                MyEntity!all()
+                !filter(e | e.fLong == p1);
+            
+            query MyEntity[] queryByString(String p1 = "Lorem ipsum") =>
+                MyEntity!all()
+                !filter(e | e.fString == p1);
+            
+            query MyEntity[] queryByDecimal(Decimal p1 = -1526.225) =>
+                MyEntity!all()
+                !filter(e | e.fDecimal == p1);
+            
+            query MyEntity[] queryByMyEnum(MyEnum p1 = MyEnum#A00) =>
+                MyEntity!all()
+                !filter(e | e.fEnum == p1);
+        '''.parse => [
+            assertNoErrors
+        ]
+    }
+    
+    // TODO: JNG-4566
+    // @Test
```

Comment:
>  `|`
> `V`
```
// TODO: JNG-4566
// @Test
```

Please use the following format for disabled tests:
```
@Test
@Disabled("JNG-4566")
```

Reply:
OK